src/dom.js

```diff
@@ -13,35 +13,241 @@
  * License for the specific language governing permissions and limitations under
  * the License.
  */
-import { JSDOM } from 'jsdom';
+
+import parse5 from 'parse5';
+import select from 'css-select';
+
+// htmlparser2 has a relatively DOM-like tree format, which we'll massage into a DOM elsewhere
+const treeAdapter = require('parse5-htmlparser2-tree-adapter');
+
+const PARSE5_OPTS = {
+  treeAdapter
+};
 
 /**
- * Parse HTML into a mutable, serializable DOM Document, provided by JSDOM.
- * @private
+ * Parse HTML into a mutable, serializable DOM Document.
+ * The DOM implementation is an htmlparser2 DOM enhanced with basic DOM mutation methods.
  * @param {String} html   HTML to parse into a Document instance
  */
-export function createDocument (html) {
-  const jsdom = new JSDOM(html, {
-    contentType: 'text/html'
-  });
-  const { window } = jsdom;
-  const document = window.document;
-  document.$jsdom = jsdom;
+export function createDocument(html) {
+  const document = parse5.parse(html, PARSE5_OPTS);
+
+  defineProperties(document, DocumentExtensions);
+
+  // Extend Element.prototype with DOM manipulation methods.
+  const scratch = document.createElement('div');
+  // Get a reference to the base Node class - used by createTextNode()
+  document.$$Node = scratch.constructor;
+  const elementProto = Object.getPrototypeOf(scratch);
+  defineProperties(elementProto, ElementExtensions);
+  elementProto.ownerDocument = document;
+
   return document;
 }
+
 /**
  * Serialize a Document to an HTML String
- * @private
  * @param {Document} document   A Document, such as one created via `createDocument()`
  */
-export function serializeDocument (document) {
-  return document.$jsdom.serialize();
+export function serializeDocument(document) {
+  return parse5.serialize(document, PARSE5_OPTS);
```

Comment:
Idea (to be validated):

if you pass the `{ sourceCodeLocationInfo: true }` option when parsing, the `link` nodes will include the location in the source. You could use that information to clean up the HTML string instead of serializing the document (and then injecting the inline styles). This **might** be faster.